namerd/core/src/main/scala/io/buoyant/namerd/DtabStore.scala

```diff
@@ -18,6 +18,12 @@ trait DtabStore {
   def create(ns: String, dtab: Dtab): Future[Unit]
 
   /**
+   * Deletes a dtab.  Returns a DtabNamespaceDoesNotExistException if the
+   * namespace does not exist.
+   */
+  def delete(ns: String): Future[Unit]
```

Comment:
Got it, thanks for the link. FWIW, k8s also gives us that functionality for free.